pkg/api/docs/definitions/dashboard_permissions.go

```diff
@@ -0,0 +1,65 @@
+package apidocs
+
+import (
+	"github.com/grafana/grafana/pkg/models"
+)
+
+// swagger:route GET /dashboards/id/{DashboardID}/permissions dashboard_permissions getDashboardPermissions
+//
+// Gets all existing permissions for the given dashboard.
+//
+// Responses:
+// 200: getDashboardPermissionsResponse
+// 401: unauthorisedError
+// 403: forbiddenError
+// 404: notFoundError
+// 500: internalServerError
+
+// swagger:route POST /dashboards/id/{DashboardID}/permissions dashboard_permissions postDashboardPermissions
+//
+// Updates permissions for a dashboard.
+//
+// This operation will remove existing permissions if they’re not included in the request.
+//
+// Responses:
+// 200: okResponse
+// 400: badRequestError
+// 401: unauthorisedError
+// 403: forbiddenError
+// 404: notFoundError
+// 500: internalServerError
+
+// swagger:parameters postDashboardPermissions updateFolderPermissions
+type PostDashboardPermissionsParam struct {
+	// in:body
+	// required:true
+	Body UpdateDashboardAclCommand
+}
+
+// swagger:response getDashboardPermissionsResponse
+type GetDashboardPermissionsResponse struct {
+	// in: body
+	Body []*models.DashboardAclInfoDTO `json:"body"`
+}
+
+// UpdateDashboardAclCommand is same as dtos.UpdateDashboardAclCommand but with swagger annotations
+// swagger:model
+type UpdateDashboardAclCommand struct {
+	// The permission items to add/update. Items that are omitted from the list will be removed.
+	Items []DashboardAclUpdateItem `json:"items"`
+}
+
+// DashboardAclUpdateItem is same as dtos.DashboardAclUpdateItem but with swagger annotations
+// swagger:model
+type DashboardAclUpdateItem struct {
```

Comment:
I have copied the model so that I can enriched it with the required swagger annotations (and I have done this also for some [other](https://github.com/grafana/grafana/blob/c659d087d8b221f8f3029a63bbb9ba0d9c0fef65/pkg/api/docs/definitions/datasources.go#L352) models).
This has the obvious drawback that it makes it harder to maintain them (keep them synced).
On the other hand, that way we don't pollute `dtos` package with swagger annotations.
I'm indecisive which is better. WDYT?

Reply:
I agree and in fact I have adopted this for [endpoints I documented later](https://github.com/grafana/grafana-enterprise/pull/2767/files#diff-b5e23ff13aa6a39cbb15e14fd2b0147f599265f22aca5c54e441eab1fcca6491)
I'll remove them and add the annotations in the original DTO

Reply:
Nice :+1: As long as we have a plan moving towards this it's fine by me. I guess the only culprit changing this might be Enterprise, but we can figure that out along the way